File: frontend/crop_logo.py
from PIL import Image
try:
    img = Image.open('src/assets/invenger logo.jpg')
    # The logo has the icon on the left. The total width is probably around 300, height 76 based on previous magick.
    # Let's crop a square from the left side.
    # We can get bounding box or just crop width = height
    width, height = img.size
    print(f"Original size: {width}x{height}")
    # The icon is on the left. Let's crop a box from (0, 0) to (height, height) assuming it's roughly square
    # or let's use the aspect ratio.
    left = 0
    top = 0
    right = height * 1.2 # slightly wider than square maybe? Actually let's just make it square: height x height
    bottom = height
    cropped_img = img.crop((left, top, height, bottom))
    cropped_img.save('public/invenger-logo-icon.jpg')
    print("Cropped successfully to", height, "x", height)
except Exception as e:
    print("Error:", e)

File: frontend/crop_logo.js
import { Jimp } from "jimp";
import path from "path";
import { fileURLToPath } from "url";

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

async function cropLogo() {
  try {
    const inputPath = path.join(__dirname, "src", "assets", "invenger logo.jpg");
    const outputPath = path.join(__dirname, "public", "invenger-logo-icon.jpg");

    // Read the image
    const image = await Jimp.read(inputPath);
    
    // The image size is e.g. 500x120. 
    // The icon is on the left. The height is the limiting factor. Let's make it a square crop.
    const height = image.bitmap.height;
    const cropWidth = height; // Square crop
    
    // Crop: x, y, w, h
    image.crop( {x: 0, y: 0, w: cropWidth, h: height } );
    
    // Resize to standard favicon sizes could be good, but just cropping is enough since browsers support any size.
    await image.write(outputPath);
    console.log("Successfully cropped logo to", cropWidth, "x", height);
  } catch (error) {
    console.error("Error cropping image:", error);
  }
}

cropLogo();
